subprojects/common/retrace/src/main/kotlin/com/avito/retrace/ProguardRetracer.kt

```diff
@@ -0,0 +1,36 @@
+package com.avito.retrace
+
+import proguard.retrace.ReTrace
+import java.io.File
+import java.io.LineNumberReader
+import java.io.PrintWriter
+import java.io.StringReader
+import java.io.StringWriter
+
+interface ProguardRetracer {
+
+    fun retrace(content: String): String
+
+    class Impl(
+        private val mappings: List<File>
+    ) : ProguardRetracer {
+
+        override fun retrace(content: String): String {
+            var result = content
+            mappings.map { ReTrace(ReTrace.STACK_TRACE_EXPRESSION, false, it) }
```

Comment:
We create a `ReTrace` for the same mapping file on each call of `retrace(content)`. It looks inefficient to read mapping file too many times

Reply:
Fixed that

Reply:
Fixed